rust/drivers/datafusion/src/lib.rs

```diff
@@ -707,39 +789,81 @@ pub struct DataFusionStatement {
+        key: Self::Option,
+        value: adbc_core::options::OptionValue,
     ) -> adbc_core::error::Result<()> {
-        todo!()
+        match key.as_ref() {
+            constants::ADBC_INGEST_OPTION_TARGET_TABLE => match value {
+                OptionValue::String(value) => {
+                    self.ingest_target_table = Some(value);
+                    Ok(())
+                }
+                _ => Err(Error::with_message_and_status(
+                    "IngestOptionTargetTable value must be of type String",
+                    Status::InvalidArguments,
+                )),
+            },
+            _ => Err(Error::with_message_and_status(
+                format!("Unrecognized option: {key:?}"),
+                Status::NotFound,
+            )),
+        }
     }
 
-    fn get_option_string(&self, _key: Self::Option) -> adbc_core::error::Result<String> {
-        todo!()
+    fn get_option_string(&self, key: Self::Option) -> adbc_core::error::Result<String> {
+        match key.as_ref() {
+            constants::ADBC_INGEST_OPTION_TARGET_TABLE => {
+                let target_table = self.ingest_target_table.clone();
+                match target_table {
+                    Some(table) => Ok(table),
+                    None => Err(Error::with_message_and_status(
+                        format!("{} has not been set", key.as_ref()),
+                        Status::NotFound,
+                    )),
+                }
+            }
+            _ => Err(Error::with_message_and_status(
+                format!("Unrecognized option: {key:?}"),
+                Status::NotFound,
+            )),
+        }
     }
 
-    fn get_option_bytes(&self, _key: Self::Option) -> adbc_core::error::Result<Vec<u8>> {
-        todo!()
+    fn get_option_bytes(&self, key: Self::Option) -> adbc_core::error::Result<Vec<u8>> {
+        Err(Error::with_message_and_status(
+            format!("Unrecognized option: {key:?}"),
+            Status::NotFound,
+        ))
     }
 
-    fn get_option_int(&self, _key: Self::Option) -> adbc_core::error::Result<i64> {
-        todo!()
+    fn get_option_int(&self, key: Self::Option) -> adbc_core::error::Result<i64> {
+        Err(Error::with_message_and_status(
+            format!("Unrecognized option: {key:?}"),
+            Status::NotFound,
+        ))
     }
 
-    fn get_option_double(&self, _key: Self::Option) -> adbc_core::error::Result<f64> {
-        todo!()
+    fn get_option_double(&self, key: Self::Option) -> adbc_core::error::Result<f64> {
+        Err(Error::with_message_and_status(
+            format!("Unrecognized option: {key:?}"),
+            Status::NotFound,
+        ))
     }
 }
 
 impl Statement for DataFusionStatement {
-    fn bind(&mut self, _batch: arrow_array::RecordBatch) -> adbc_core::error::Result<()> {
-        todo!()
+    fn bind(&mut self, batch: arrow_array::RecordBatch) -> adbc_core::error::Result<()> {
+        self.bound_record_batch.replace(Some(batch));
```

Comment:
Using `Option::replace` instead:
```suggestion
        self.bound_record_batch.replace(batch);
```